feat: fulfillment app boilerplate improvements

tolerants opened the pull request:
- [x] Optimized FA boilerlate
- [x] Drop unused files in FA boilerlate
- [x] Renamed `StopEventsInitializer` into `CaptureEventsInitializer`
- [x] Changed `oryx-app` rendering, so now it's not rendering header and footer.
- [x] Added header and footer into experience pages. 
- [x] Improvements from https://github.com/spryker/oryx/pull/837
closes: [HRZ-89640](https://spryker.atlassian.net/browse/HRZ-89640), [HRZ-89641](https://spryker.atlassian.net/browse/HRZ-89641), [HRZ-89553](https://spryker.atlassian.net/browse/HRZ-89553)


[HRZ-89640]: https://spryker.atlassian.net/browse/HRZ-89640?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HRZ-89641]: https://spryker.atlassian.net/browse/HRZ-89641?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HRZ-89553]: https://spryker.atlassian.net/browse/HRZ-89553?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
Fix

Commit:
Fix

Commit:
Removed console.logs.

Commit:
Merge remote-tracking branch 'origin/development' into feat/hrz-89553-fa-boilerplate

# Conflicts:
#	apps/fulfillment/src/app.ts
#	libs/template/presets/storefront/experience/pages/cart-page.ts
#	libs/template/presets/storefront/experience/pages/checkout-page.ts
#	libs/template/presets/storefront/experience/pages/contact-page.ts
#	libs/template/presets/storefront/experience/pages/login-page.ts
#	libs/template/presets/storefront/experience/pages/order-confirmation-page.ts

Commit:
Dev merge

Commit:
Formatting

Commit:
Fix styles

Commit:
Fix

Commit:
Fix

Commit:
Fix

Commit:
Merge remote-tracking branch 'origin/development' into feat/hrz-89553-fa-boilerplate

# Conflicts:
#	libs/template/presets/storefront/experience/header.ts

Commit:
Fixed capture event initializer.

Commit:
Fix

Commit:
Fix formatting

Commit:
Merge remote-tracking branch 'origin/development' into feat/hrz-89553-fa-boilerplate

Commit:
Fixed contact and register pages.

Commit:
Formatting.

Commit:
Add force check into facets e2e test.

Commit:
Formatting

Commit:
Fix b2b header.

tobi-or-not-tobi reviewed (changes requested):
There are a few important things related to the layout rendering I like to change in this PR. They're all related to the `type` we introduced to `OryxAppComponent`. I believe that's not the right thing to do, as we have everything we need in Oryx to manage layout for various screen sizes already. 

Here are the pieces:
- FA theme should only supports 1 breakpoint, we should add the small screen breakpoint rather than reuse the `defaultBreakpoints`
- The FA theme should not add all breakpoints for all sizes, but stick to sm only. Most likely it's cleaner to create an fa design token file, or bring them directly into the theme instead
- There's a bug in the currently used `backofficeNgTokens` that is not applying the container width correctly, as it's been wrapped in `layout` field.
- In order to leverage the small screen only, and apply the centered layout because of that, we should ideally leverage experience pages rather than connect components directly to the routes. If not easily possible, we could reuse layout styles in the various components to leverage the layout.

Commit:
Merge remote-tracking branch 'origin/development' into feat/hrz-89553-fa-boilerplate

# Conflicts:
#	apps/fulfillment/src/app.ts
#	libs/template/presets/fulfillment/app.ts
#	libs/template/presets/storefront/experience/pages/checkout-page.ts

Commit:
Formatting.

tobi-or-not-tobi committed:
refactor(presets): clean fa theme (#837)

We like to improve the fulfilment app theming to leverage standard
theming options for the max width of the application.

- [x] Move `backoffice-ng` design-token content into `backoffice`.
- [x] Remove `backoffice-ng` design-token. 
- [x] Create small-backoffice design-token set.
- [x] FA theme should use 2 breakpoints and layout should behave
differently for these two:
  - 530px - layout should take the whole width
  - 768px - layout is fixed (530px) and centred
- [x] Search box: fix float behaviour - for now the searchbox takes
almost whole width of the view port
- [x] Modal: fix width for fulfillment
- [x] Add experience pages:
  - [x] Login Page 
  - [x] Picking lists Page
  - [x] Warehouse Selection

closes [HRZ-90261](https://spryker.atlassian.net/browse/HRZ-90261),
[HRZ-90260](https://spryker.atlassian.net/browse/HRZ-90260)

[HRZ-90261]:
https://spryker.atlassian.net/browse/HRZ-90261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[HRZ-90260]:
https://spryker.atlassian.net/browse/HRZ-90260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---------

Co-authored-by: Alexander Kovalenko <alexander.kovalenko@spryker.com>

Commit:
Update presets dependencies.

Commit:
Updated .env.template.

supproduction commented on libs/template/presets/storefront/experience/pages/registration-page.ts:
```diff
@@ -58,69 +49,94 @@ export const registrationPage: ExperienceComponent = {
                   {
-                    colSpan: 2,
-                    margin: '10px 0 0',
+                    layout: CompositionLayout.Grid,
```
Please, don't use `CompositionLayout` we are going to deprecate it

tolerants replied:
Thanks, fixed.

Commit:
Fixed CR comments.

Commit:
Formatting.

tobi-or-not-tobi commented on apps/fulfillment/.env.template:
```diff
@@ -1 +1,11 @@
-ORYX_FULFILLMENT_CLIENT_ID=fulfillment-client-local
+# The url where the backend is hosted.
+ORYX_FULFILLMENT_BACKEND_URL=
```
It would be nice if we could align this env var with the one we use in storefront. reason behind is that this would allow us to reuse launchpad logic to create env vars. If you agree, let's create a ticket for this.

tolerants replied:
Let's discuss it on catchup.

tolerants replied:
It's impossible as long as we gonna have the same variable for storefront and fulfillment on github.

tobi-or-not-tobi commented on libs/template/application/oryx-app/oryx-app.component.ts:
```diff
@@ -11,9 +11,13 @@ export class OryxAppComponent extends LitElement {
   protected override render(): TemplateResult {
     return html`
-      <oryx-composition uid="header"></oryx-composition>
+      ${featureVersion < '1.2'
```
https://spryker.slack.com/archives/C029XPSBXC6/p1697630001287519

tolerants replied:
Fixed.

tobi-or-not-tobi commented on libs/template/presets/fulfillment/app.ts:
```diff
@@ -43,13 +49,18 @@ export function fulfillmentFeatures(
     new BapiAuthComponentsFeature(),
     { resources: fulfillmentResources },
-    new FulfillmentRootFeature(config?.fulfillmentRoot),
+    featureVersion < '1.2'
```
`<= '1.2'

tolerants replied:
Fixed.

Commit:
CR fixes.

Commit:
Fixed picking version.

Commit:
Revert modal fix.

Commit:
Revert searchbox fix.

Commit:
Merge remote-tracking branch 'origin/development' into feat/hrz-89553-fa-boilerplate

# Conflicts:
#	libs/template/presets/storefront/experience/pages/cart-page.ts
#	libs/template/presets/storefront/experience/pages/category-page.ts
#	libs/template/presets/storefront/experience/pages/checkout-page.ts
#	libs/template/presets/storefront/experience/pages/login-page.ts
#	libs/template/presets/storefront/experience/pages/product-page.ts
#	libs/template/presets/storefront/experience/pages/registration-page.ts
#	libs/template/presets/storefront/experience/pages/search-page.ts

Commit:
Formatting.

Commit:
Trigger ci

Commit:
Merge remote-tracking branch 'origin/development' into feat/hrz-89553-fa-boilerplate

# Conflicts:
#	libs/template/presets/storefront/experience/pages/product-page.ts

Commit:
Formatting.

Commit:
Fixed picking version.

Commit:
Fixed backoffice design tokens.

Commit:
Fixed FA.

Commit:
Fix.

tobi-or-not-tobi commented on apps/fulfillment/index.html:
```diff
@@ -34,9 +34,14 @@
       href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500;600;700&display=swap"
       rel="stylesheet"
     />
+    <style>
+      :root {
+        font-size: 16px;
+      }
+    </style>
```
For now this is OK. 

Capturing our conversation: 
The problem with the font-size is related to the `DefaultThemeMetaInitializer` we have. This is creating a hard-coded font-size of `14px` to the root of the document. This is not responsive currently - it must be `16px` on small screen.
While all styling typically belongs to the `<oryx-app>` element, the font-size must be set on the root, in order to make `rem` unit work. This is the only exception. Other stuff cascades down from the `oryx-app` element.

Going forward, I can think of 2 solutions:
1. make metaService responsive, observing the viewport and change the hardcoded font-size based on small vs large - bad idea, the meta service should not be bothered with this
2. introduce a additional section of document CSS variables (we only need font-size for now, but I've seen other cases where it could be useful). we then add the font-size here, and we can use that CSS var in the meta service.
